### PR TITLE
[ENG-9064][1/2] EAS Update: a basic debugging guide, try 2

### PR DESCRIPTION
**Pull request by quinlanj**
# Why
Same idea as https://github.com/expo/expo/pull/23046 but rewrites the basic debugging guide with @kbrandwijk 's [suggestion](https://github.com/expo/expo/pull/23046#issuecomment-1603984947). 

# How

The basic debugging guide is meant to be concrete and simple. It should take users no longer than 10 minutes to go through the entire thing. It points users to the Deployments page and uses that as a starting point for all debugging solutions. 

basic-guide

This PR moves the current debugging guide to `debugging-advanced.mdx`.

# Test Plan

- [ ] manually tested

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/eas-update/debug.mdx [hunk 1]
```diff
@@ -1,69 +1,87 @@
...
+This guide shows how to verify our configuration so that we can find the source of problems like an app not showing a published update. It's important to tell the current state of our app at any given time, so we built EAS Update with this in mind. Once we know which updates are running on which builds, we can make changes so that our apps are in the state we expect.
 
-## expo-updates configuration
+## Go to the Deployments page
 
-The expo-updates library runs inside an end-user's app and makes requests to an update server to get the latest update.
+The EAS website has a [Deployments page](https://expo.dev/accounts/[account]/projects/[project]/deployments) that shows the current state of our app. The term _deployment_ refers to a group of builds and their updates. If we've made builds and updates with EAS, you can see your project's status on the website in the Deployments tab.
```
Comment (wschurman) [inline docs/pages/eas-update/debug.mdx:14]:
```suggestion
The EAS website has a [Deployments page](https://expo.dev/accounts/[account]/projects/[project]/deployments) that shows the current state of our app. The term _deployment_ refers to a group of builds and their corresponding updates. If we've made builds and updates with EAS, we can see our project's status on the website in the Deployments tab.
```

(fwiw, I don't really like the `we` voice for docs in general, but that's outside the scope of this PR. this suggestion just adheres to a consistent voice)

File: docs/pages/eas-update/debug.mdx [hunk 2]
```diff
@@ -1,69 +1,87 @@
...
 
-### Inspecting expo-updates configuration after prebuild
+<ImageSpotlight
+  alt="Deployment has no updates"
+  src="/static/images/eas-update/deployments-wrong-channel.png"
+  style={{ maxWidth: 600 }}
+/>
 
-Whenever we run `eas build`, the `npx expo prebuild` command is run on our project on EAS' servers to unpack the **android** and **ios** directories that contain native files. This makes it so EAS Build can build any project, whether it includes the native files or not.
+If the deployment channel is unexpected, it means your build was not built with the correct channel. To fix this, [configure your channel](/#configure-channel).
```
Comment (wschurman) [inline docs/pages/eas-update/debug.mdx:36]:
```suggestion
If the deployment channel is unexpected, it means our build was not built with the correct channel. To fix this, [configure our channel](/#configure-channel).
```

File: docs/pages/eas-update/debug.mdx [hunk 3]
```diff
@@ -1,69 +1,87 @@
...
 
-In each, we expect to see configuration for the EAS Update URL and the runtime version. Here are the properties we'd expect to see in each file:
+<ImageSpotlight
+  alt="Deployment has no updates"
+  src="/static/images/eas-update/deployments-wrong-runtime.png"
+  style={{ maxWidth: 600 }}
+/>
 
-**AndroidManifest.xml**
+If the deployment runtime version is unexpected, it means your build was not built with the correct runtime version. To fix this, [configure your runtime version](/#configure-runtime-version).
```
Comment (wschurman) [inline docs/pages/eas-update/debug.mdx:46]:
```suggestion
If the deployment runtime version is unexpected, it means our build was not built with the correct runtime version. To fix this, [configure our runtime version](/#configure-runtime-version).
```

File: docs/pages/eas-update/debug.mdx [hunk 4]
```diff
@@ -1,69 +1,87 @@
...
 
-```xml
-...
-<key>EXUpdatesRuntimeVersion</key>
-<string>your-runtime-version-here</string>
-<key>EXUpdatesURL</key>
-<string>https://u.expo.dev/your-project-id-here</string>
-...
-```
+If the deployment has an unexpected branch, you will need to [map your channel to the correct branch](/#map-channel-to-branch).
```
Comment (wschurman) [inline docs/pages/eas-update/debug.mdx:56]:
```suggestion
If the deployment has an unexpected branch, we will need to [map our channel to the correct branch](/#map-channel-to-branch).
```

File: docs/pages/eas-update/debug.mdx [hunk 5]
```diff
@@ -1,69 +1,87 @@
...
 
-To debug the state of EAS Update in our project, we'll need to look at multiple spots in the system. Below is a diagram of how EAS Update works and the spots that are useful to inspect when finding the root cause of an issue. In the sections following, we'll inspect and verify these spots and more.
+<ImageSpotlight
+  alt="Deployment has no updates"
+  src="/static/images/eas-update/deployments-no-updates.png"
+  style={{ maxWidth: 600 }}
+/>
 
-<ImageSpotlight alt="Map of debugging spots" src="/static/images/eas-update/debug-map.png" />
+The displayed deployment does not have any updates. To fix this, [publish an update to the branch](/#publish-update).
```
Comment (wschurman) [inline docs/pages/eas-update/debug.mdx:66]: I like this voice much more than the `we` fwiw.

File: docs/pages/eas-update/debug.mdx [hunk 6]
```diff
@@ -1,69 +1,87 @@
...
+  src="/static/images/eas-update/deployments-no-branch.png"
+  style={{ maxWidth: 600 }}
+/>
+
+The displayed deployment has the correct channel, but it is not linked to a branch. To fix this, [map the channel to the correct branch](/#map-channel-to-branch).
+
+### Missing deployment
 
-#### Configuring channels
+If your deployment is not displayed, this means your build is not configured properly with EAS Update. To fix this, [configure your channel](/#configure-channel) and [configure your runtime version](/#configure-runtime-version).
```
Comment (wschurman) [inline docs/pages/eas-update/debug.mdx:80]:
```suggestion
If your deployment is not displayed, this means our build is not configured properly with EAS Update. To fix this, [configure our channel](/#configure-channel) and [configure our runtime version](/#configure-runtime-version).
```

File: docs/pages/eas-update/debug.mdx [hunk 7]
```diff
@@ -83,41 +101,25 @@ To verify that a build has a specific channel, make sure that in **eas.json**, t
...
-<Terminal
-  cmd={[
-    '# eas channel:create [channel-name]',
-    '',
-    '',
-    '# Example',
-    'eas channel:create production',
-  ]}
-/>
+To verify your runtime version, make sure your app config (**app.json**/**app.config.js**) has a `runtimeVersion` property:
```
Comment (wschurman) [inline docs/pages/eas-update/debug.mdx:106]:
```suggestion
To verify your runtime version, make sure our app config (**app.json**/**app.config.js**) has a `runtimeVersion` property:
```

File: docs/pages/eas-update/debug.mdx [hunk 8]
```diff
@@ -83,41 +101,25 @@ To verify that a build has a specific channel, make sure that in **eas.json**, t
...
+  "expo": {
+    "runtimeVersion": {
+      "policy": "sdkVersion"
+    }
+  }
+}
+```
 
-There is a link that is defined by the developer between a channel and a branch. When a channel and branch are linked, an app with a channel will get the most recent compatible update on the linked branch.
+By default, it is `{ "policy": "sdkVersion" }`, but you can change your runtime to [use a different policy or a specific version](/eas-update/runtime-versions).
```
Comment (wschurman) [inline docs/pages/eas-update/debug.mdx:118]:
```suggestion
By default, it is `{ "policy": "sdkVersion" }`, but we can change our runtime to [use a different policy or a specific version](/eas-update/runtime-versions).
```

File: docs/pages/eas-update/debug.mdx [hunk 9]
```diff
@@ -129,166 +131,15 @@ If the channel is not linked to the branch we expect, we can change the link wit
...
-Every branch contains a list of updates. When a build makes a call for an update, we find the channel of the build, then the branch linked to that channel. Once the branch is found, EAS will return the most recent compatible update on that branch. A build and an update are compatible when they share the same runtime version and platform.
-
-To inspect which updates are on a branch, we can run:
-
-<Terminal
-  cmd={['# eas branch:view [branch-name]', '', '', '# Example', 'eas branch:view production']}
-/>
-
-The output of this command will show us a list of updates and their runtime versions and platforms. From this list, we should be able to figure out which update should apply to a given build, by matching the build's runtime version and platform to update's runtime version and platform. The most recent update that is compatible will be available for a build to download and execute.
+If your branch is not listed, you can create a new branch with `eas branch:create`
```
Comment (wschurman) [inline docs/pages/eas-update/debug.mdx:134]:
```suggestion
If your branch is not listed, we can create a new branch with `eas branch:create`
```

**Review by wschurman — CHANGES_REQUESTED** [review]
My comments are mostly to adhere to the `we` voice in the doc consistently. I personally don't like using this voice or reading this voice as a documentation consumer, but I understand why we do it and changing it it outside the scope of this PR I think.

File: docs/pages/eas-update/debug.mdx [hunk 10]
```diff
@@ -1,69 +1,87 @@
...
-sidebar_title: Debug
+sidebar_title: Basic
 description: Learn how to debug EAS Update.
 ---
 
 import ImageSpotlight from '~/components/plugins/ImageSpotlight';
 import { Terminal } from '~/ui/components/Snippet';
 
-It's important to tell the current state of our app at any given time. We built EAS Update with this in mind. Once we know which updates are running on which builds, we can make changes so that our apps are in the state we expect and desire. This guide sets out to show how we can verify our EAS Update and expo-updates configuration so that we can find the source of problems like an app not showing a published update.
+This guide shows how to verify our configuration so that we can find the source of problems like an app not showing a published update. It's important to tell the current state of our app at any given time, so we built EAS Update with this in mind. Once we know which updates are running on which builds, we can make changes so that our apps are in the state we expect.
```
Comment (kbrandwijk) [inline docs/pages/eas-update/debug.mdx:10]: Multiple personality 'we' in this sentence as well.

File: docs/pages/eas-update/debug.mdx (hunk 2 above)
Comment (kbrandwijk) [inline docs/pages/eas-update/debug.mdx:36]: Emphasize that a rebuild of the app is necessary after changing this (common mistake).

File: docs/pages/eas-update/debug.mdx [hunk 11]
```diff
@@ -1,69 +1,87 @@
...
 
-Finally, make sure that `expo-updates` is included in **package.json**. If it's not, run:
+<ImageSpotlight alt="Map of debugging spots" src="/static/images/eas-update/debug-map.png" />
 
-<Terminal cmd={['$ npx expo install expo-updates']} />
+### Unexpected channel
 
-### Inspecting expo-updates configuration after prebuild
+<ImageSpotlight
+  alt="Deployment has no updates"
```
Comment (kbrandwijk) [inline docs/pages/eas-update/debug.mdx:31]: Fix various alt texts on the different screenshots throughout.

File: docs/pages/eas-update/debug.mdx (hunk 5 above)
Comment (kbrandwijk) [inline docs/pages/eas-update/debug.mdx:66]: Other reason is that the runtime version _of the update_ doesn't match that of the build. This is common enough to note here as well.

File: docs/pages/eas-update/debug.mdx (hunk 6 above)
Comment (kbrandwijk) [inline docs/pages/eas-update/debug.mdx:80]:
```suggestion
If your deployment is not displayed, this means your build is not configured properly for EAS Update. To fix this, [configure your channel](/#configure-channel) and [configure your runtime version](/#configure-runtime-version).
```

Maybe mention as well to verify the general EAS Update configuration (updates.url)?

File: docs/pages/eas-update/debug.mdx [hunk 12]
```diff
@@ -1,69 +1,87 @@
...
+### Missing deployment
 
-#### Configuring channels
+If your deployment is not displayed, this means your build is not configured properly with EAS Update. To fix this, [configure your channel](/#configure-channel) and [configure your runtime version](/#configure-runtime-version).
+
+## Solutions
+
+### Configure channel
 
 To verify that a build has a specific channel, make sure that in **eas.json**, there is a channel property:
```
Comment (kbrandwijk) [inline docs/pages/eas-update/debug.mdx:86]:
```suggestion
To verify that a build has a specific channel, make sure that your build profile in **eas.json** has a channel property:
```

File: docs/pages/eas-update/debug.mdx (hunk 9 above)
Comment (kbrandwijk) [inline docs/pages/eas-update/debug.mdx:134]:
```suggestion
If our branch is not listed, we can create a new branch with `eas branch:create`
```

File: docs/pages/eas-update/debug.mdx [hunk 13]
```diff
@@ -129,166 +131,15 @@ If the channel is not linked to the branch we expect, we can change the link wit
...
-        "buildConfiguration": "Debug"
-      },
-      "channel": "preview_debug"
-    }
-  }
-}
-```
+After publishing, the output will display the branch and the runtime version. This information can help us verify that we're creating an update with the configuration we expect.
 
 ## Mitigation steps
```
Comment (kbrandwijk) [inline docs/pages/eas-update/debug.mdx:144]: This is now a bit disconnected from the previous section. Any 'root cause of the issue' we might have found, was a configuration one, not an issue with the actual update up to this point.

File: docs/pages/eas-update/debug.mdx [hunk 14]
```diff
@@ -336,6 +187,6 @@ Since "update 3" is now the most recent update on the "production" branch, all u
 
 While this will prevent all new users from seeing the bad update, users who've already received the bad update will run it until they can download the latest update. Since mobile networks are not always able to download the most recent update, sometimes users may run a bad update for a long time. When viewing error logs for our app, it's normal to see a lingering long tail of errors as our users' apps get the most recent update or build. We'll know we solved the bug when we see the error rate decline dramatically; however, it likely will not disappear completely if we have a diverse user base across many locations and mobile networks.
 
-## Wrap up
+## Next steps
 
-Still having issues with EAS Update? Provide us with a reproduction repo in our [forums](https://forums.expo.dev/c/expo-application-services/56). Also, feel free to ask in the **#update** channel on our [community Discord](https://chat.expo.dev/), or [contact us](https://expo.dev/contact) directly.
+Read the [advanced guide](/eas-update/debug-advanced) to debug EAS Update when the basic steps don't work.
```
Comment (kbrandwijk) [inline docs/pages/eas-update/debug.mdx:192]:
```suggestion
Read the [advanced guide](/eas-update/debug-advanced) to debug EAS Update when the basic steps don't resolve your issue.
```

File: docs/pages/eas-update/debug.mdx (hunk 7 above)
Comment (kbrandwijk) [inline docs/pages/eas-update/debug.mdx:106]:
```suggestion
To verify our runtime version, we make sure our app config (**app.json**/**app.config.js**) has a `runtimeVersion` property:
```

**Review by wschurman — APPROVED** [review]

**Pull request merged**